bundles/org.openhab.binding.sbus/src/main/java/org/openhab/binding/sbus/handler/SbusRgbwHandler.java

```diff
@@ -0,0 +1,239 @@
+     * @return an int array [R, G, B, W] each in [0..255]
+     */
+    private int[] hsbToRgbw(HSBType hsbType) {
+        if (hsbType == null) {
+            throw new IllegalArgumentException("HSBType cannot be null.");
+        }
+
+        // Convert HSBType to standard RGB [0..255]
+        PercentType[] rgb = ColorUtil.hsbToRgbPercent(hsbType);
+        // Convert each channel from 0..100 to 0..255
+        int r = (int) Math.round(rgb[0].floatValue() * 2.55);
+        int g = (int) Math.round(rgb[1].floatValue() * 2.55);
+        int b = (int) Math.round(rgb[2].floatValue() * 2.55);
+
+        // Determine the white component as the min of R, G, B
+        int w = Math.min(r, Math.min(g, b));
+
+        // Subtract W from each
+        r -= w;
+        g -= w;
+        b -= w;
+
+        return new int[] { r, g, b, w };
+    }
+
+    /**
+     * Converts an RGBW array ([R, G, B, W]) back to an openHAB HSBType.
+     *
+     * We add the W channel back into R, G, and B, then clamp to [0..255].
+     * Finally, we create an HSBType via fromRGB().
+     *
+     * @param rgbw an int array [R, G, B, W] each in [0..255]
+     * @return an HSBType (hue [0..360], saturation/brightness [0..100])
+     */
+    private HSBType rgbwToHsb(int[] rgbw) {
+        if (rgbw == null || rgbw.length < 4) {
+            throw new IllegalArgumentException("rgbw must be non-null and have 4 elements: [R, G, B, W].");
+        }
+
+        int r = rgbw[0];
+        int g = rgbw[1];
+        int b = rgbw[2];
+        int w = rgbw[3];
+
+        // Restore the combined R, G, B
+        int rTotal = r + w;
+        int gTotal = g + w;
+        int bTotal = b + w;
+
+        // Clamp to [0..255]
+        rTotal = Math.min(255, Math.max(0, rTotal));
+        gTotal = Math.min(255, Math.max(0, gTotal));
+        bTotal = Math.min(255, Math.max(0, bTotal));
+
+        // Convert back to an HSBType via fromRGB
+        HSBType hsbType = HSBType.fromRGB(rTotal, gTotal, bTotal);
+
+        return hsbType;
+    }
+
+    /**
+     * Checks if any RGBW value is greater than 0.
+     *
+     * @param rgbw an int array [R, G, B, W] each in [0..255]
+     * @return true if any value is greater than 0, false otherwise
+     */
+    private boolean isAnyRgbwValueActive(int[] rgbw) {
+        if (rgbw == null || rgbw.length < 4) {
+            return false;
+        }
+        for (int value : rgbw) {
+            if (value > 0) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected void initializeChannels() {
+        int switchChannelCount = 0;
+
+        // Validate all channel configurations
+        for (Channel channel : getThing().getChannels()) {
+            SbusChannelConfig channelConfig = channel.getConfiguration().as(SbusChannelConfig.class);
+            String channelTypeId = channel.getChannelTypeUID().getId();
+            if ("color-channel".equals(channelTypeId)) {
```

Comment:
These channel strings can be extract as constant in the `BindingConstatnts.java`
Same for channels below.